### DISCUSSION TITLE
Code doesn't get passed if using the the prompt_actions when using the Telescope integration

### DISCUSSION THREAD

**Discussion by Andrej-Marsic** [CopilotC-Nvim/CopilotChat.nvim#288] [Q&A]
I assume in some cases it's the visual selection that gets passed to the prompt, but if one tries to use it through the telescope integration, the selection disappears and nothing gets passed (or the current line at most)

**Comment by Coding-Journey** [comment]
Weirdly the default Nvim behaviour of get visual selection updates _after_ you escape the selection, so I find that if I select and unselect the thing I want to chat about, it'll work. However it'd be great for this to be improved

**Reply by lukebarton** [reply]
Useful workaround, annoying problem though!

**Comment by deathbeam** [comment]
We can definitely cancel selection earlier, and we should, I think this is actually a bug and should be an issue not discusson

**Comment by pidgeon777** [comment]
If someone finds it to be useful, here is my custom `selection` function:

```lua
        selection = function(source)
          local select = require("CopilotChat.select")

          -- HACK: We detect the if normal or visual mode is active first,
          -- otherwise if left as per LazyVim default, visual_sel will always
          -- be checked first, and return the selection between '< and '> marks,
          -- even if no visual selection is currently active, thus:
          -- - buffer_sel wouldn't be done.
          -- - selected text could be wrong if text is added or removed to the
          --   original buffer => this could lead to an error!
          -- TODO: Report issue to LazyVim repo?
          local mode = vim.api.nvim_get_mode().mode
          -- vim.notify("Mode: " .. mode)
          if mode == "n" then
            local buffer_sel = select.buffer(source)
            -- vim.notify("Buffer selection: " .. vim.inspect(buffer_sel))
            return buffer_sel
          elseif mode == "v" then
            local visual_sel = select.visual(source)
            -- vim.notify("Visual selection: " .. vim.inspect(visual_sel))
            return visual_sel
          end
        end,
```

**Comment by deathbeam** [comment]
i changed how leaving visual mode works (as we are forcing visual mode leave as workaround beacuse of it), could someone with telescope maybe retry if it works with default config now? Maybe it will
